source/PreferencesPanel.cpp

```diff
@@ -583,27 +586,49 @@ void PreferencesPanel::DrawPlugins()
 	const Color &back = *GameData::Colors().Get("faint");
 	const Color &medium = *GameData::Colors().Get("medium");
 	const Color &bright = *GameData::Colors().Get("bright");
+	const Color &error = *GameData::Colors().Get("bright error");
+
+	const Sprite *box[2] = { SpriteSet::Get("ui/unchecked"), SpriteSet::Get("ui/checked") };
 
 	const int MAX_TEXT_WIDTH = 230;
 	Table table;
 	table.AddColumn(-115, {MAX_TEXT_WIDTH, Truncate::MIDDLE});
-	table.SetUnderline(-120, 120);
+	table.SetUnderline(-120, 100);
 
 	int firstY = -238;
-	table.DrawAt(Point(-130, firstY));
+	// Table is at -110 while checkbox is at -130
+	table.DrawAt(Point(-110, firstY));
 	table.DrawUnderline(medium);
 	table.Draw("Installed plugins:", bright);
 	table.DrawGap(5);
 
 	const Font &font = FontSet::Get(14);
+
 	for(const auto &plugin : GameData::PluginAboutText())
 	{
 		pluginZones.emplace_back(table.GetCenterPoint(), table.GetRowSize(), plugin.first);
 
 		bool isSelected = (plugin.first == selectedPlugin);
 		if(isSelected || plugin.first == hoverPlugin)
 			table.DrawHighlight(back);
-		table.Draw(plugin.first, isSelected ? bright : medium);
+		bool pluginEnabled = Plugins::IsEnabled(plugin.first);
+		double rowHeight = table.GetRowBounds().Height();
+		double thirdHeight = rowHeight / 3;
+		Point checkboxPos = table.GetRowBounds().TopLeft();
+		checkboxPos.X() -= font.Height() / 2;
+		checkboxPos.Y() += rowHeight / 2;
+
+		SpriteShader::Draw(box[pluginEnabled], checkboxPos);
+		Point zoneDimension = Point(20., thirdHeight * 2.);
+		Point zoneOffset = checkboxPos + Point(0., thirdHeight - 3);
+		Rectangle zoneBounds = Rectangle(zoneOffset, zoneDimension);
+		AddZone(zoneBounds, [&]() { Plugins::TogglePlugin(plugin.first); });
+		if(Plugins::HasChanged(plugin.first))
+			table.Draw(plugin.first, error);
```

Comment:
Definitely don't use an error color when the plugin is free of errors. If you want to reflect that the plugin is currently disabled, or still currently active, use the colors that are normally used in that scenario. The checkbox encapsulates the change that will take effect.

Reply:
The color can be renamed to “plugins changed” instead of error if you like.

There’s three colors at play.

1. Standard enabled (checked/bright)
2. Standard disabled (unchecked/dim)
3. A selection has changed from current runtime requiring restart (currently red text).

If a user changes plugin runtime the code highlights which plugins were changed and show a restart message.  If user decides to revert their selection the restart message goes away because the selection matches the runtime.  Currently, there’s no standard way of displaying this.

I can rename the variable and keep the same color or suggest a color and we can see how it looks.

Might be helpful to understand intent by building and messing with the UI.

Reply:
I'm saying that the color of the plugin name should only depend on the enabled-ness of the plugin at the time the game was started, as that is the enabled-ness of the plugin if the player enters their ship after leaving this menu.

The plugin checkbox state reflects the enabled-ness following the next restart.

Reply:
That was how the original PR worked.  Since making the changes with the red text (I am flexible on the color).  It is convenient if I or another user changes our mind and decides to not toggle plugins being enabled.

By reverting selections to match startup the need to restart goes away.

For a large number of plugins (greater than 10) one does not need to memorize their previous selection.

Reply:
Why do you mention memorizing? The scheme I propose requires no memorization by the user:
1) draw the plugin name as either `bright` or `dim`, based on the initial state of the plugin, when the game booted.
2) draw the checkbox as selected or unselected, based on the current state of the plugin.

Using an error color when the plugin has no errors in it is misleading. The game does not _currently_ evaluate plugin definitions for errors, but it _will_.

Reply:
I don't think you understand the current implementation just by reading the code.  You need to launch the game and see how it behaves because it already works as you describe.  The warning state is when the user makes enable/disable changes.  It needs to be highlighted in case the user changes their mind.

I encourage you to:
1. launch the game.
2. Disable a few plugins (not all)
3. Restart  the game.  You will see dim and normal text based on current state.
4. Make some changes and you will see how they highlight changes you've made from the current runtime.

Memory is definitely an issue when you have state tracked for a dozen plus plugins and then change their runtime for next restart.  The coloring solves the problem user would have to remember their selection if they want to undo their changes.

Reply:
I renamed the code reference from `error` to `pluginReloadRequired`.  This is hopefully more clear as to the color's purpose and not to be an error.

Reply:
![image](https://user-images.githubusercontent.com/875669/193474878-4b36b935-f6f8-4911-8382-7fd87c6d3fb9.png)

Here's a screenshot of your suggestion.  The current selected plugin is hard to tell.  You have to move your mouse a few times to get the full view of what plugins changed in terms of user clicking check boxes.

Reply:
Pushed the suggestion, though I think from a usability perspective it's more difficult to change plugin selections without some kind of button to revert back to initial selection.  e.g. restore defaults.